file_hierarchy.test.js

```diff
@@ -0,0 +1,165 @@
+const { directory, file, cleanAll } = require('./file_hierarchy');
+const { join, sep } = require('path');
+const { tmpdir } = require('os');
+const fs = require('fs');
+
+const mymkdtempSync = fs.mkdtempSync;
+fs.mkdtempSync = jest.fn((elem) => mymkdtempSync(elem));
+const mockFunction = jest.fn(() => { });
+
+
+describe('directory', () => {
+
```

Comment:
```suggestion
  it("should call the mkdirtemp function and the function delivered to the directory function", () => {
    //GIVEN
    const call = join(tmpdir(), "xanthippe");

    //WHEN
    workingDirectory = "foo";
    directory("testdir", () => mockFunction());

    //THEN
    expect(fs.mkdtempSync).toHaveBeenCalledWith(call);
    expect(mockFunction).toHaveBeenCalled();
  });
```

Can you add this test to your testsuite and run it again?
Does it still succeed?
If not, why?